server/src/main/java/org/opensearch/indices/replication/OngoingSegmentReplications.java

```diff
@@ -147,6 +148,16 @@ void startSegmentCopy(GetSegmentFilesRequest request, ActionListener<GetSegmentF
      */
     CopyState prepareForReplication(CheckpointInfoRequest request, FileChunkWriter fileChunkWriter) throws IOException {
         final CopyState copyState = getCachedCopyState(request.getCheckpoint());
+        if (copyState.getCheckpoint()
+            .getLatestSupportedCodec()
+            .equalsIgnoreCase(request.getCheckpoint().getLatestSupportedCodec()) == false) {
+            // cancel(request.getTargetAllocationId(), "Requested unsupported codec version");
+            logger.trace("Requested unsupported codec version {}", request.getCheckpoint().getLatestSupportedCodec());
+            throw new CancellableThreads.ExecutionCancelledException(
```

Comment:
It will be good to have one integration test to verify behavior on target around handling this exception and verifying shard is not failed.

Reply:
Same as above.